go/storage/mkvs/db/badger/badger.go

```diff
@@ -163,6 +179,76 @@ func (d *badgerNodeDB) sanityCheckNamespace(ns common.Namespace) error {
 	return nil
 }
 
+// Assumes metaUpdateLock is held when called.
+func (d *badgerNodeDB) cleanMultipartLocked(removeNodes bool) error {
+	var version uint64
+
+	if d.multipart {
+		version = d.multipartVersion
+	} else {
+		version = d.meta.getMultipartVersion()
+	}
+	if version == 0 {
```

Comment:
Hm so a non-zero version is used as a multipart indicator? Why is a separate `d.multipart` flag needed then? Can it happen that `d.multipart == true && d.multipartVersion == 0`?

Reply:
I had just the flag initially, then left it in for slightly easier reasoning. But yeah, it's redundant.